spec/console_spec.rb

```diff
@@ -429,48 +440,47 @@
         expect(current_subject).to receive_message_chain(:gets, :chomp) { cancel_input }
 
         current_subject.destroy_account
-
-        expect(File.exist?(OVERRIDABLE_FILENAME)).to be false
       end
     end
   end
 
   describe '#show_cards' do
-    let(:cards) { [{ number: 1234, type: 'a' }, { number: 5678, type: 'b' }] }
+    let(:cards) { [Card.new(:usual), Card.new(:capitalist)] }
 
     it 'display cards if there are any' do
-      current_subject.instance_variable_set(:@current_account, instance_double('Account', card: cards))
-      cards.each { |card| expect(current_subject).to receive(:puts).with("- #{card[:number]}, #{card[:type]}") }
+      current_subject.instance_variable_set(:@current_account, instance_double(Account, card: cards))
+      cards.each { |card| expect(Card).to receive(:puts).with("- #{card.number}, #{card.type}") }
       current_subject.show_cards
     end
 
     it 'outputs error if there are no active cards' do
-      current_subject.instance_variable_set(:@current_account, instance_double('Account', card: []))
-      expect(current_subject).to receive(:puts).with(ERROR_PHRASES[:no_active_cards])
+      current_subject.instance_variable_set(:@current_account, instance_double(Account, card: []))
+      expect(Card).to receive(:puts).with(ERROR_PHRASES[:no_active_cards])
       current_subject.show_cards
     end
   end
 
   describe '#create_card' do
+    let(:test_account) { Account.new('Anton', 'Anton43', 'Anton43', 43) }
+
     context 'with correct outout' do
       it do
         CREATE_CARD_PHRASES.each { |phrase| expect(current_subject).to receive(:puts).with(phrase) }
-        current_subject.instance_variable_set(:@card, [])
-        current_subject.instance_variable_set(:@current_account, current_subject)
+        current_subject.instance_variable_set(:@current_account, test_account)
         allow(current_subject).to receive(:accounts).and_return([])
         allow(File).to receive(:open)
         expect(current_subject).to receive_message_chain(:gets, :chomp) { 'usual' }
 
-        current_subject.create_card
+        current_subject.send(:input_card_type)
```

Comment:
write correct test if u need to use send

Reply:
fixed to defolt test